src/content/en/2019/http2.md

```diff
@@ -10,22 +10,34 @@ last_updated: 2019-11-04T12:00:00+00:00:00
 
-Despite HTTP being text-based, the reality is that it was rarely used to transport text, at least in it's raw format. While it was true that HTTP headers were still text, the payloads themselves often were not. Text files like HTML, JS and CSS are usually [compressed](./compression) for transport into a binary format using gzip, brotli or similar and non-text files like images, videos... etc. are served in their own formats. The whole HTTP message is then often wrapped in HTTPS to encrypt the messages for security reasons. So, the web had basically moved on from text-based transport a long time ago, but HTTP had not. One reason for this stagnation was because it was so difficult to introduce any breaking changes to such a ubiquitous protocol like HTTP (previous efforts had tried and failed). Many routers, firewalls and other middleboxes understand HTTP and would react badly to major changes to it. Upgrading them all to support a new version is simply not possible.
+The protocol seemed simple, but that also meant certain limitations - the main one being that HTTP was basically synchronous: once an HTTP request had been sent, the whole TCP connection was basically off limits for anything else until the response had come back, been read and processed. This was incredibly inefficient and required multiple TCP connections (browsers typically use 6) to allow a limited form of parallelization.
 
-In 2009 Google announced they were working on an alternative to the text-based HTTP called SPDY. This would take advantage of the fact that HTTP messages were often encrypted in HTTPS which prevents them being read and interfered with en route. Google controlled one of the most popular browsers (Chrome) and some of the most popular websites (Google, YouTube, Gmail...etc.) - so both ends of the connection. Google's idea was to pack HTTP messages into a proprietary format, send them across the internet, and then unpacked them on the other side. The proprietary format (SPDY) was binary-based rather than text-based which solved some of the main performance problems with HTTP/1.1 by allowing more efficient use of a single TCP connection, negating the need to open the 6 connections that had become the norm under HTTP/1.1. By using SPDY in the real world they were able to prove it was more performant for real users, and not just because of some lab-based experimental results. After rolling out SPDY to all Google websites, other servers and browser started implementing it, and then it was time to standardize this proprietary format into an internet standard and thus HTTP/2 was born.
```

Comment:
A key part of being able to roll out SPDY and HTTP/2 was that Google controlled both sides of the connection, but see you cut that piece of text. Don't think this is worth including?

Reply:
It felt quite lengthy, so maybe just include a more concise explanation for this part?

Reply:
Fair enough. You're right it is quite lengthy so let's see if others have any comments and, if not, then go with what you did here.

Reply:
More content the merrier IMO. It's on-point so LGTM.